### PR TITLE
chore: upstream classical tactic

### PR DESCRIPTION
**Pull request by semorrison**

### CODE REVIEW

**Comment by fgdorais** [comment]
Does this really depend on #241? I think it could be made to do without.

**Comment by semorrison** [comment]
> Does this really depend on #241? I think it could be made to do without.

I was not anticipating this being nearly as slow; I thought when I started that both of these PRs would be merged 30 minutes-hours later, but here we are 60 days later.

Yes, the only "dependence" is through a test.

**Comment by semorrison** [comment]
This PR is not essential for anything I'm doing in the short term; if you'd like to disentangle them please do!

File: Std/Tactic/Classical.lean [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+def bar : Bool := by
+  classical
+  have := ∀ p, decide p -- uses the classical instance
+  exact decide (0 < 1) -- uses the decidable instance
+```
+Note that (unlike lean 3) `classical` is a scoping tactic - it adds the instance only within the
+scope of the tactic.
+-/
+-- FIXME: using ppDedent looks good in the common case, but produces the incorrect result when
+-- the `classical` does not scope over the rest of the block.
```
Comment (fgdorais) [inline Std/Tactic/Classical.lean:46]: Can this be fixed in this PR?

**Review by fgdorais — APPROVED** [review]
Looks good to me modulo the question about the FIXME.

**Review by joehendrix — APPROVED** [review]

**Pull request merged**